astroid/interpreter/_import/spec.py

```diff
@@ -143,15 +143,45 @@ def find_module(
                 type=ModuleType.C_BUILTIN,
             )
 
+        if submodule_path is not None:
+             search_paths = list(submodule_path)
+         else:
+             search_paths = sys.path
+
+         suffixes = (".py", ".pyi", importlib.machinery.BYTECODE_SUFFIXES[0])
+         for entry in search_paths:
+             package_directory = os.path.join(entry, modname)
+             for suffix in suffixes:
+                 package_file_name = "__init__" + suffix
+                 file_path = os.path.join(package_directory, package_file_name)
+                 if cached_os_path_isfile(file_path):
+                     return ModuleSpec(
+                         name=modname,
+                         location=package_directory,
+                         type=ModuleType.PKG_DIRECTORY,
+                     )
+             for suffix, type_ in ImportlibFinder._SUFFIXES:
+                 file_name = modname + suffix
+                 file_path = os.path.join(entry, file_name)
+                 if cached_os_path_isfile(file_path):
```

Comment:
@jacobtylerwalls Bit afraid of this change... The `3.3.x` branch doesn't have it so I'm not sure if this introduce more issues.

But if you approve I guess we can just try to make it work.